website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,255 @@
+    expr: |
+      max by (name, condition, endpoint) (cluster_operator_conditions{name="version", condition="Upgradeable", endpoint="metrics"} == 0)
+    for: 60m
+    labels:
+      severity: warning
+```
+
+This alert fires if one or more operators have not reported their `Upgradeable`
+condition as true in more than an hour.  The alert has a clear name and
+informative summary and description annotations.  The timeline is appropriate
+for allowing the operator a chance to resolve the issue automatically, avoiding
+the need to alert an administrator.
+
+##### Info Alerts
+
+TL/DR: Info level alerts represent situations an administrator should be aware
+of, but that don't necessarily require any action.  Use these sparingly, and
+consider instead reporting this information via Kubernetes events.
+
+Example info alert: [MultipleContainersOOMKilled](https://github.com/openshift/cluster-monitoring-operator/blob/79cdf68/assets/cluster-monitoring-operator/prometheus-rule.yaml#L326-L338)
+
+```yaml
+- alert: MultipleContainersOOMKilled
+  annotations:
+    description: Multiple containers were out of memory killed within the past
+      15 minutes. There are many potential causes of OOM errors, however issues
+      on a specific node or containers breaching their limits is common.
+      summary: Containers are being killed due to OOM
+  expr: sum(max by(namespace, container, pod) (increase(kube_pod_container_status_restarts_total[12m]))
+    and max by(namespace, container, pod) (kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}) == 1) > 5
+  for: 15m
+  labels:
+    namespace: kube-system
+    severity: info
+```
+
+This alert fires if multiple containers have been terminated due to out of
+memory conditions in the last 15 minutes.  This is something the administrator
+should be aware of, but may not require immediate action.
+
+### Alerts, Metrics and Recording Rules Tests
+
+1. Add tests for alerts that validate that:
+   - Each alert includes all mandatory fields.
+   - Each `runbook_url` link is valid.
+   - Each alert that includes a `pod` or a `container` also includes the `namespace`.
+2. Add e2e tests that inspect the alerts during upgrade and make sure that the alerts don’t fire when they shouldn’t (Zero noise).
+3. Add tests for metrics/recording rules that validate that:
+   - Metric / Recording rule exists
+   - Metric / Recording rule value is as expected
+   - Metric / Recording rule name follows the best practices guideline
+
+### Test Plan
+
+Automated tests enforcing acceptance criteria for critical alerts and basic style linting will be added to the [openshift/origin](https://github.com/openshift/origin) end-to-end test suite.
+The monitoring team will work with anyone shipping existing critical alerts that
+don't meet these criteria in order to resolve the issue before enabling the tests.
```

Comment:
This one does not fit well here. We should not add anything that is specific to OCP.